devlib/module/sched.py

```diff
@@ -209,12 +209,22 @@ class SchedDomain(SchedProcFSNode):
     def __init__(self, nodes):
         super(SchedDomain, self).__init__(nodes)
 
-        obj_flags = set()
-        for flag in list(SchedDomainFlag):
-            if self.flags & flag.value == flag.value:
-                obj_flags.add(flag)
+        # Recent kernels now have a space-separated list of flags instead of a
+        # packed bitfield
+        flags = self.flags
+        if isinstance(flags, str):
+            flags = flags.split()
+            def has_flag(flags, flag):
+                return flag in flags
```

Comment:
Unfortunately this change (kernel side I mean) also removed a flag which caused a shift in value for all the others. The point was that moving forward only the flag *name* would have value to external users, as the value cannot be used anymore (there was a writable interface which has now been removed).

I'm regretting the Enum here as I don't see a nice way to dynamically discover available flags while keeping it backwards-compatible. We *could* "fake" an Enum with `.name` being the name read above and `.value` either returning a bogus value + issuing a warning or raising an exception altogether.

Also, worth noting I'm the one who made those kernel changes, so send me your alcohol bill...

Reply:
Is there really an issue in the end ? If the value has no use at all on new kernels, it does not really matter that much that it is bogus. If we are scared someone will use it to debug in kernel issues or that it comes back via e.g. the need to decode an ftrace event field, we can either put a "bomb" value that will raise when used or maybe simply compute the right value if the kernel outputs the names in bitfield order with a known starting point

Reply:
Ack for the value not being an issue for newer kernels

As for runtime discovery of flag names, I don't have any better idea if we're to keep the Enum.